### PR TITLE
feat: automatic env detection for hubble login

### PR DESCRIPTION
**Pull request by Andrei997**
Ref #95

### CODE REVIEW

**Comment by bwanglzu** [comment]
great, let me play with it on your branch

**Comment by bwanglzu** [comment]
seems it's not detected in jupyter notebook, another question, what if we succesfully detect is_notebook=True and it is jupyter lab? i remember the `notebook_login` won't work in jupyter lab.

**Comment by Andrei997** [comment]
> seems it's not detected in jupyter notebook, another question, what if we succesfully detect is_notebook=True and it is jupyter lab? i remember the `notebook_login` won't work in jupyter lab.

It definitely should be detected in jupyter notebook, are you checking it properly ? :D

**Comment by Andrei997** [comment]
> seems it's not detected in jupyter notebook, another question, what if we succesfully detect is_notebook=True and it is jupyter lab? i remember the `notebook_login` won't work in jupyter lab.

About jupyter lab, that is a good point, how do you handle that in fine-tuner ?

**Comment by bwanglzu** [comment]
@Andrei997 we claim in the docs the `notebook_login` support jupyter notebook and google colab

**Comment by Andrei997** [comment]
> @Andrei997 we claim in the docs the `notebook_login` support jupyter notebook and google colab

I see, I'll check if there is a way to detect lab vs notebook environments.

**Comment by hanxiao** [comment]
Please raise PR in this docs accordingly https://docs.jina.ai/jina-ai-cloud/login/

File: tests/unit/utils/test_notebook.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from hubble.utils.notebook import is_notebook
+
+
+async def test_is_notebook():
+    is_notebook_env = is_notebook()
+    assert is_notebook_env is False
```
Comment (alaeddine-13) [inline tests/unit/utils/test_notebook.py:6]:
```suggestion
    assert not is_notebook()
```

**Review by alaeddine-13 — COMMENTED** [review]
does it make sense to remove the public notebook login method to avoid confusing users ?

**Comment by Andrei997** [comment]
We now attempt to identify jupyter lab environments

**Comment by Andrei997** [comment]
@bwanglzu Can you please check if this works well for fine-tuner? Thank you.

**Comment by Andrei997** [comment]
The widget login should now be supported in Jupyter Lab as well! @bwanglzu could you have a look to make sure, I've tested it in both Jupyter Notebook and Jupyter lab and it seems to work fine.

**Comment by bwanglzu** [comment]
@Andrei997 i'll try out in 1 hour great job!

File: hubble/__init__.py [hunk 2]
```diff
@@ -55,9 +56,16 @@ def arg_wrapper(*args, **kwargs):
     return arg_wrapper
 
 
-def login(**kwargs):
-    """This function guide user to browser log-in and get token."""
-    asyncio.run(Auth.login(**kwargs))
+def login(interactive=None, **kwargs):
```
Comment (hanxiao) [inline hubble/__init__.py:59]:
```suggestion
def login(interactive: Optional[bool] = None, **kwargs):
```

File: requirements.txt [hunk 3]
```diff
@@ -6,4 +6,5 @@ filelock
 pathspec
 cryptography
 docker
-pyyaml
\ No newline at end of file
+pyyaml
+psutil
```
Comment (hanxiao) [inline requirements.txt:10]: why?

**Pull request merged**